testing/trino-product-tests/src/main/java/io/trino/tests/product/jdbc/TestKerberosConstrainedDelegationJdbc.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.tests.product.jdbc;
+
+import com.google.inject.Inject;
+import com.google.inject.name.Named;
+import io.trino.tempto.BeforeMethodWithContext;
+import io.trino.tempto.ProductTest;
+import io.trino.tempto.kerberos.KerberosAuthentication;
+import io.trino.tests.product.TpchTableResults;
+import org.ietf.jgss.GSSCredential;
+import org.ietf.jgss.GSSException;
+import org.ietf.jgss.GSSManager;
+import org.ietf.jgss.Oid;
+import org.testng.annotations.Test;
+
+import javax.security.auth.Subject;
+
+import java.security.Principal;
+import java.security.PrivilegedActionException;
+import java.security.PrivilegedExceptionAction;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.util.Properties;
+
+import static com.google.common.base.Throwables.throwIfInstanceOf;
+import static com.google.common.base.Throwables.throwIfUnchecked;
+import static io.trino.tempto.assertions.QueryAssert.assertThat;
+import static io.trino.tempto.query.QueryResult.forResultSet;
+import static io.trino.tests.product.TestGroups.JDBC_KERBEROS_CONSTRAINED_DELEGATION;
+import static org.ietf.jgss.GSSCredential.DEFAULT_LIFETIME;
+import static org.ietf.jgss.GSSCredential.INITIATE_ONLY;
+import static org.ietf.jgss.GSSName.NT_USER_NAME;
+
+public class TestKerberosConstrainedDelegationJdbc
+        extends ProductTest
+{
+    private static final String KERBEROS_OID = "1.2.840.113554.1.2.2";
+    @Inject
+    @Named("databases.presto.jdbc_url")
+    String jdbcUrl;
+
+    @Inject
+    @Named("databases.presto.kerberos_principal")
+    private String kerberosPrincipal;
+
+    @Inject
+    @Named("databases.presto.kerberos_keytab")
+    private String kerberosKeytab;
+
+    private GSSManager gssManager;
+    private KerberosAuthentication kerberosAuthentication;
+
+    @BeforeMethodWithContext
+    public void setUp()
+    {
+        this.gssManager = GSSManager.getInstance();
+        this.kerberosAuthentication = new KerberosAuthentication(kerberosPrincipal, kerberosKeytab);
+    }
+
+    @Test(groups = JDBC_KERBEROS_CONSTRAINED_DELEGATION)
+    public void testConstrainedDelegationKerberos()
```

Comment:
Can we test `CREATE` and `INSERT` type of queries here ?

Reply:
CTAS test added